modules/uncertainty/R/ensemble.R

```diff
@@ -207,28 +207,28 @@ write.ensemble.configs <- function(defaults, ensemble.samples, settings, model,
     return(list(runs = NULL, ensemble.id = NULL))
   }
   
+  # See if we need to write to DB
+  write.to.db <- as.logical(settings$database$bety$write)
   # Open connection to database so we can store all run/ensemble information
-  if (write.to.db) {
-    con <- try(PEcAn.DB::db.open(settings$database$bety), silent = TRUE)
-    if (inherits(con, "try-error")) {
-      con <- NULL
-    } else {
-      on.exit(PEcAn.DB::db.close(con))
-    }
-  } else {
+  con <- try(PEcAn.DB::db.open(settings$database$bety), silent = TRUE)
+  on.exit(try(PEcAn.DB::db.close(con), silent = TRUE))
+  
+  # If we fail to connect to DB then we set to NULL
+  if (inherits(con, "try-error")) {
     con <- NULL
   }
+
   
   # Get the workflow id
-  if ("workflow" %in% names(settings)) {
+  if (!is.null(settings$workflow$id)) {
```

Comment:
won't this throw an error, rather than return NULL, if `settings$workflow` doesn't exist

Reply:
`[[` subset syntax will, but `$` sub-setting returns `NULL`. This is actually one of the reasons I typically prefer `[[` in my own code -- I like when things fail early, with loud and clear errors, rather than silently propagating `NULL`. But in this case, it's appropriate.

``` r
  l <- list(a = 3)
  l$x$y$z
#> NULL
```

<sup>Created on 2019-06-27 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

Reply:
Actually, turns out `[[` only throws an error if you do a relatively uncommon kind of deep indexing `[[c(...)]]`. If you just index one-at-a-time, it will silently propagate the `NULL` through everything.

``` r
  l <- list(a = 3)
  l$x$y
#> NULL
  l[["x"]][["y"]]
#> NULL
  l[[c("x", "y")]]
#> Error in l[[c("x", "y")]]: no such index at level 1
```

<sup>Created on 2019-06-27 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>